webapp/drive/static/scripts/container.ts

```diff
@@ -0,0 +1,51 @@
+export abstract class Containable<T> extends HTMLElement {
+    // The HTMLElement to be collected.
+    data: T
+
+    abstract update(data: T): void;
+    abstract show(): void;
+    abstract hide(): void;
+};
+
+export class Container<T> {
+    // Makes it easier to add elements or
+    // update already existing ones.
+    content = new Array<Containable<T>>();
+    parent: HTMLElement;
+    elementType: new (p: HTMLElement, i: number) => Containable<T>;
+
+    constructor(parent: HTMLElement, elementType: new (p: HTMLElement, i: number) => Containable<T>) {
+        this.parent = parent;
+        this.elementType = elementType;
+    }
+
+    setContent(content: T[]) {
+        // Displays the given list of objects
+        for (var i=0; i < content.length; i++)
```

Comment:
never use var, use always const or let
```suggestion
        for (let i=0; i < content.length; i++)
```

Reply:
i am quite sure i need var in this case because the variable i is also used later (outside the for loop), and using "let" would cause it to be out of scope

Reply:
then, create the variable using let before the for cycle
```typescript
let i;
for (i = 0; i < content.length; i++)
```